### ISSUE TITLE
Imageloader loads from internet before disk cache

### ISSUE THREAD

**Issue by tymerica** [nostra13/Android-Universal-Image-Loader#1034]
When in airplane mode the phone loads all the photos from disk cache, when connected to the internet it tries to reload them from the internet. It gives the error Image cant be decoded whether or not im connected to the internet, then gives a false url with a size with the error. I am using the display image method. 

here is the error
(obviously basewebsite isnt the website but it is not giving the proper url, it has the wrong url but the image still loads just not from the cache as long as i have a network connection)

Image can't be decoded [https://basewebsite.com_1440x640]

These are my configs

  DisplayImageOptions defaultOptions = new DisplayImageOptions.Builder()

                .cacheInMemory(true)

                .cacheOnDisk(true)

                .resetViewBeforeLoading(true)

               .bitmapConfig(Bitmap.Config.RGB_565)

                .build();
        ImageLoaderConfiguration config = new ImageLoaderConfiguration.Builder(getApplicationContext())
                .defaultDisplayImageOptions(defaultOptions)
                .memoryCache(new WeakMemoryCache())
               
               .threadPoolSize(5)
            
                .build();
        ImageLoader.getInstance().init(config);

**Comment by nostra13** [comment]
https://github.com/nostra13/Android-Universal-Image-Loader/wiki/Useful-Info Item 9

**Comment by tymerica** [comment]
@nostra13 But my question is, why is it only loading from disk cache while in airplane mode, otherwise it reloads them from the internet? It just doesnt load them from the cache while it has an internet connection. It used to, before i updated the JAR.

**Comment by nostra13** [comment]
It should load from disk cache if there is cached image.
Are you sure UIL loads from disk if no internet and loads from network if it has internet?

**Comment by tymerica** [comment]
@nostra13  
Yes, if i have 3g, the images stay blank for a few seconds while trying to load them and making data requests and eventually they load in (obviously i saved them to the disk cache before testing this), and if the phone is in airplane mode, they all load instantly from the cache. I tested it by storing them as specific sizes, if the size is less than 200 x 200 it is from cache otherwise it is from the internet. While it has internet all photos are their original sizes, while it does not have internet, they are all the scaled down versions stored in the cache.

**Comment by tymerica** [comment]
Or rather, any internet connection, i just used 3g to demonstrate that I could see the images loading too slow to possibly be from the disk cache.

**Comment by BellaSecret** [comment]
maybe you can ask to this website www.smspromedia.com
